lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -417,6 +417,12 @@ SILCombiner::optimizeApplyOfConvertFunctionInst(FullApplySite AI,
   if (SubstCalleeTy->hasArchetype() || ConvertCalleeTy->hasArchetype())
```

Comment:
I feel like this restriction is completely unnecessary.

Reply:
May be, but this is not the part of this bug-fix patch.

Reply:
Please try removing it in another patch and see what breaks :-)